torch/_refs/__init__.py

```diff
@@ -1770,6 +1805,15 @@ def full_like(
 zeros_like = partial(full_like, fill_value=False)
 
 
+def scalar_tensor(
+    a: NumberType,
+    *,
+    dtype: Optional[torch.dtype] = None,
+    device: Optional[torch.device] = None,
+) -> TensorLikeType:
+    return prims.scalar_tensor(a, dtype=dtype, device=device)
```

Comment:
This should determine the appropriate dtype and device and the prim should require them